minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/ObservationFromRayImplementation.java

```diff
@@ -198,52 +198,46 @@ else if (mop.typeOfHit == RayTraceResult.Type.ENTITY)
         json.add("LineOfSight", jsonMop);
     }
 
-    static RayTraceResult findEntity(Vec3d eyePos, Vec3d lookVec, double depth, RayTraceResult mop, boolean includeTiles)
+    public static RayTraceResult findEntity(Vec3d eyePos, Vec3d lookVec, double depth, RayTraceResult mop, boolean includeTiles)
     {
         // Based on code in EntityRenderer.getMouseOver()
+
+        // Don't return entities through other blocks!
         if (mop != null)
             depth = mop.hitVec.distanceTo(eyePos);
-        Vec3d searchVec = eyePos.addVector(lookVec.xCoord * depth, lookVec.yCoord * depth, lookVec.zCoord * depth);
+
+        // Scale lookVec by the depth and don't assume it has been normalized
+        lookVec = lookVec.scale(depth / lookVec.lengthVector());
```

Comment:
can lengthvector be 0?

Reply:
In theory it could be! But MC's lengthVector is normalized to length 1